File: Python/Q21.py
# Q21.py - Program to Find GCD of Two Numbers
def gcd(x, y):
    while y != 0:
        x, y = y, x % y
    return x

num1 = int(input("Enter first number: "))
num2 = int(input("Enter second number: "))
print(f"GCD is {gcd(num1, num2)}")

File: Java/Q21.java
import java.util.Scanner;

public class GCDofTwoNumbers {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.print("Enter first number: ");
        int num1 = sc.nextInt();

        System.out.print("Enter second number: ");
        int num2 = sc.nextInt();

        while (num2 != 0) {
            int temp = num2;
            num2 = num1 % num2;
            num1 = temp;
        }

        System.out.println("GCD is: " + num1);
        sc.close();
    }
}
